### DISCUSSION TITLE
Moving to Org-based billing

### DISCUSSION THREAD

**Discussion by kiwicopple** [supabase/supabase#17061] [Changelog]
We’re fixing the billing system at Supabase - moving from “project-based” to “organization-based”. We should have started with this model, but I wasn’t wise enough to know that when we started. We need to make these changes to roll out Preview Environments / Branching. It also includes:

- [long-requested](https://github.com/orgs/supabase/discussions/6681) project transfers between organizations
- An extra 1GB egress on the Free Tier
- Consolidated invoices
- Self-serve Team plan
- Updates for branching
- No more “upfront” charges for Database Compute Addons

[See all changes in the blog post](https://supabase.com/blog/organization-based-billing)

## Free plan

First, and most importantly - there is only one change that affects the free plan, and that is a good one for you: you now an extra 1GB of egress. 

| Usage Item | Old plan (per project) | New plan (org based) |
| --- | --- | --- |
| Egress | 4GB - (2GB Database + 2GB Storage) | 5GB across Database + Storage  |
| Database Space | 500MB  | 500MB |
| Storage Space | 1GB  | 1GB |
| Monthly Active Users | 50K | 50K |
| Edge Function Invocations | 500K | 500K |
| Edge Function Count | 10 | 10 |
| Realtime Message Count | 2 million | 2 million |
| Realtime Peak Connections | 200 | 200 |
|  | 2 free projects | 2 free orgs (1 free database per org) |

On top of an extra 1GB of egress for free, now that egress is unified across your org it means that if you aren’t using Supabase Storage, you get even more Database Egress (5GB instead of 2GB previously)

If you are currently running 2 free projects however, ***this does require some work from you***. Because we are now working on an Org-level, instead of Projects, you will need to:

1. Create a new “Free org”
2. Transfer one of your free projects into the newly-created org

This should be done before the end of October, but don’t worry - we’ll give you frequent comms and clear instructions once the change has been rolled out (4th Sept).

## Other changes

We’ve made a lot of improvements to the billing system. Read the full announcement on our [blog](https://supabase.com/blog/organization-based-billing) or dive into the related [docs](https://supabase.com/docs/guides/platform/org-based-billing) for more details.

## Help, my bill increased!

This is a major change, and we've tried to design it in a way that's cheaper for everyone. If your bill has increased as a result of this change, that's not our intention. Please submit a Support ticket [on the dashboard](https://supabase.com/dashboard/support/new?category=billing&subject=Organization%20based%20billing) and we'll figure out a solution.

## Please keep this discussion on topic

We welcome any questions/feedback about this change, but please keep this discussion focused only on this change! It's important for those who want to learn more or are confused. If you have something off-topic, please open a new discussion or join an existing discussion

**Comment by AdditionAddict** [comment]
I don't get how bill doesn't increase. First paid plan $25 now goes to $25 + $10 compute? (Opened a ticket incase I'm misunserstanding)

**Reply by GaryAustin1** [reply]
There is a $10 credit per month built into the plan per org.

![image](https://github.com/supabase/supabase/assets/54564956/c7c13f4c-f716-4bd2-a5a7-7f053965781c)

**Reply by kevcodez** [reply]
What Gary said - the $10 in Compute Credits will cover one Starter Compute (default) project. Launching additional projects or upgrading compute will incur compute costs. Each project is a dedicated server and database.

Some examples:

Running a single project on the Pro plan:
```
+ $25 Pro Plan
+ $10 for one project on Starter Compute
- $10 Compute Credits
=> $25 per month (like before)
```

Running three project on the Pro plan:
```
+ $25 Pro Plan
+ $30 for one project on Starter Compute
- $10 Compute Credits
=> $45 per month (would have been $75 before)
```

Running two projects with different compute size on the Pro plan:
```
+ $25 Pro Plan
+ $15 for one project on Small Compute
+ $10 for one project on Starter Compute
- $10 Compute Credits
=> $40 per month (would have been $55 before)
```

One thing to keep in mind is that we no longer allow mixing paid and non-paid projects in a single organization. If you'd like to have one project on the free plan and one project on the Pro plan, you'd have to create two different organizations.

**Reply by Akash187** [reply]
### My Problem
This new billing has major flows.
1. _Now inviting a developer becomes Org-level so everyone will get access to the Production database regardless of whether they need it or not._
2. Leaking Staging Credential or Edge function URL can affect Production Ingress.

### Possible Solution
1. I would recommend a cheaper plan (Pro Lite $15) which we can use for Staging or other less intensive and non-production work.
2. Project-level Invitation for members.

### My Approach
Right now I only need Production to be on the **pro plan** so I prefer using different organizations for Staging and Production. If required will move Staging to **pro plan**. It will cost 15$ more but better security.

### Final Conclusion
I don't think mixing Stating and Production billing is really a safe approach. Decoupling them seems a better approach for **security** and **peace of mind**.

**Reply by kevcodez** [reply]
User management has always been on the organization level and has not changed with organization-based billing at all.

If you have a Pro plan organization and launch a production and staging project, the staging project is cheaper to have on the Pro plan than before (previously two projects on Pro plan would be $50, now it's $35).

If you want to separate staging and production in terms of user management, you can create two organizations right now, but this was the same case with the previous billing system.

We'll eventually introduce more fine-grained permissions that will allow to configure per-project permissions and solve this - there's no timeline for this though. Billing should not be confused with permission management, these easily get mixed up and they are two separate things.

We're aware of the need for a more fine-grained permission control, so it's on the horizon 👍 .

**Reply by Akash187** [reply]
Thanks, @kevcodez So you mean it is broken from the start?  Anyway, I would like to see a project-based invitation. I am really impressed with Supabase and how rapidly you guys are pushing new features. Hope this will be available soon.

**Reply by Akash187** [reply]
@kevcodez After thinking about it for a day I don't feel confident about the org-level invitation. It should be a project-level invitation.

**Reply by kevcodez** [reply]
For the time being, you'll have to create multiple orgs with a single project to achieve that.

**Reply by bombillazo** [reply]
Project-level permission per team members is a must for collaborating with external devs (heck, even internal devs). We run our different environment projects (dev, staging, prod) for simplicity, and it makes sense they are in the same org. But now, since any team member has access to any and all projects, we'd have to manage separate orgs anyway for projects, making things harder for us.  This breaks the point of having organizations in the first place.

**Reply by kevcodez** [reply]
The whole permission system has not changed with organization-based billing at all. Permissions were always set on an organization-level, even before the billing change. As stated previously, we are aware of the need for project-based permissions and already have plans to support it in the future. 

I see that you started a separate feature request for this, which is great - https://github.com/orgs/supabase/discussions/18927

Please keep any comments/input for project-based permissions to that feature request 🙏

**Reply by Akash187** [reply]
That is exactly what I am trying to tell them. This compromises the security and privacy of the project.

Hey @bombillazo I read your medium article [I cannot fully recommend Supabase (yet)](https://bombillazo.medium.com/why-i-cannot-fully-recommend-supabase-yet-f8e994201804)and also commented on it. Very well written. I guess it a time to add this org-level billing mess also.

**Reply by kevcodez** [reply]
@Akash187 Please keep this discussion civil and on-topic. Once more, organization-based billing has not touched the permission system at all. There was no change. Feel free to upvote the [feature request](https://github.com/orgs/supabase/discussions/18927) from bombillazo, if you'd like to see it in the future.

If you have any further input for the permission system, comment in the feature request to keep this discussion on-topic for organization-based billing (permission system is completely independent).
